Project/backend/codebase/graph_creator/router.py

```diff
@@ -0,0 +1,168 @@
+import logging
+import os
+import uuid
+
+from fastapi import APIRouter, Depends
+from fastapi import UploadFile, File, HTTPException
+
+from graph_creator.dao.graph_job_dao import GraphJobDAO
+from graph_creator.schemas.graph_job import GraphJobCreate
+
+router = APIRouter()
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+# Endpoint for uploading PDF documents
+@router.post("/upload/")
+async def upload_pdf(
+    file: UploadFile = File(...), graph_job_dao: GraphJobDAO = Depends()
+):
+    """
+    Uploads a PDF document.
+
+    Args:
+        file (UploadFile): PDF document to be uploaded.
+        graph_job_dao (GraphJobDAO):
+
+    Returns:
+        dict: A dictionary containing the filename and status.
+            id (uuid.UUID): Id of the uploaded file
+            filename (str): Name of the uploaded file.
+            location (str): Location of the uploaded file.
+            status (str): Status message of the file.
+
+    Raises:
+        HTTPException: If the uploaded file type is not valid (not PDF).
+    """
+
+    # Check if the uploaded file type is correct
+    if file.content_type != "application/pdf":
+        raise HTTPException(status_code=400, detail="Uploaded file is not a PDF.")
+
+    # Define the directory for saving files
+    documents_directory = os.path.join(
+        os.path.dirname(os.path.dirname(os.path.abspath(__file__))), ".media/documents"
+    )
+    if not os.path.exists(documents_directory):
+        os.makedirs(documents_directory)
+    logger.info(documents_directory)
+
+    # Save file
+    file_path = os.path.join(documents_directory, file.filename)
+    with open(file_path, "wb") as f:
+        f.write(file.file.read())
+    logger.info(file_path)
+
+    graph_job = GraphJobCreate(
+        name=file.filename, location=file_path, status="Document uploaded"
+    )
+    # Check if graph job exists
+    existing_graph_job = await graph_job_dao.get_graph_job_by_name(graph_job.name)
+
+    # If graph job exists raise error
+    if existing_graph_job:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Graph job with name '{graph_job.name}' has already been added.",
+        )
+
```

Comment:
We should allow users to upload files with same name and handle this accordingly.